src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JavaJenkinsBuildPlanCreator.java

```diff
@@ -0,0 +1,73 @@
+package de.tum.in.www1.artemis.service.connectors.jenkins;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.w3c.dom.Document;
+
+import de.tum.in.www1.artemis.config.Constants;
+import de.tum.in.www1.artemis.service.util.XmlFileUtils;
+
+@Profile("jenkins")
+@Component
+public class JavaJenkinsBuildPlanCreator implements JenkinsXmlConfigBuilder {
+
+    private static final Logger log = LoggerFactory.getLogger(JavaJenkinsBuildPlanCreator.class);
+
+    private static final String REPLACE_TEST_REPO = "#testRepository";
+
+    private static final String REPLACE_ASSIGNMENT_REPO = "#assignmentRepository";
+
+    private static final String REPLACE_GIT_CREDENTIALS = "#gitCredentials";
+
+    private static final String REPLACE_ASSIGNMENT_CHECKOUT_PATH = "#assignmentCheckoutPath";
+
+    private static final String REPLACE_PUSH_TOKEN = "#secretPushToken";
+
+    private static final String REPLACE_ARTEMIS_NOTIFICATION_URL = "#notificationsUrl";
+
+    private static final String REPLACE_NOTIFICATIONS_TOKEN = "#jenkinsNotificationToken";
+
+    private String artemisNotificationUrl;
+
+    @Value("${artemis.jenkins.secret-push-token}")
+    private String pushToken;
+
+    @Value("${artemis.jenkins.git-credentials}")
+    private String gitCredentialsKey;
+
+    @Value("${server.url}")
+    private String ARTEMIS_BASE_URL;
+
+    @Value("${artemis.jenkins.notification-token}")
+    private String NOTIFICATION_TOKEN;
+
+    @PostConstruct
+    public void init() {
+        this.artemisNotificationUrl = ARTEMIS_BASE_URL + "/api" + Constants.NEW_RESULT_RESOURCE_PATH;
+    }
+
+    @Override
+    public Document buildBasicConfig(URL testRepositoryURL, URL assignmentRepositoryURL) {
+        final var resourcePath = Path.of("build", "jenkins", "java", "config.xml");
+        final var replacements = Map.of(REPLACE_TEST_REPO, testRepositoryURL.toString(), REPLACE_ASSIGNMENT_REPO, assignmentRepositoryURL.toString(), REPLACE_GIT_CREDENTIALS,
+                gitCredentialsKey, REPLACE_ASSIGNMENT_CHECKOUT_PATH, Constants.ASSIGNMENT_CHECKOUT_PATH, REPLACE_PUSH_TOKEN, pushToken, REPLACE_ARTEMIS_NOTIFICATION_URL,
+                artemisNotificationUrl, REPLACE_NOTIFICATIONS_TOKEN, NOTIFICATION_TOKEN);
+
+        return XmlFileUtils.readXmlFile(resourcePath, replacements);
+    }
+
+    @Override
+    public Document buildBasicConfig(URL testRepositoryURL, URL assignmentRepositoryURL, boolean isSequential) {
+        throw new NotImplementedException("Sequential Jenkins builds not yet supported for Java!");
```

Comment:
What if `false` is passed in for `isSequential`? Shouldn't the method above be called then?

Reply:
Yes, I wanted to implement the whole method in a future PR, but maybe it's better to still add the call to the above method for now.

Reply:
> What if `false` is passed in for `isSequential`? Shouldn't the method above be called then?

Yeah, probably safer to still call the above method. Changed that